File: js-pr/56/javascript/code-quality-readability-maintainability/boolean_checker_29.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function checkBooleanValue(value) {
    if (value === null || value === undefined) {
        return null;
    }
    
    if (typeof value === 'boolean') {
        return value;
    }
    
    if (typeof value === 'number') {
        return Boolean(value);
    }
    
    return false;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/boolean_checker_29.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def check_boolean_value(value):
    if value is None:
        return None
    if isinstance(value, bool):
        return value
    if isinstance(value, (int, float)):
        return bool(value)
    return False
# {/fact}